algraph.py

```diff
@@ -0,0 +1,54 @@
+
+class ALGraph:
+    """adjacency_list graph"""
+
+    def __init__(self):
+        self.adjacency_list = {}
+        
```

Comment:
Prefer keeping track of vertices explicitly -- if so, suggest adding functions `addVert(self, vert)` and `addEdge(self, edge)` that are called by `addVerts()` and `addEdges()` respectively, and allow the constructor to optionally take verts (as a list) and edges (as a list or tupl of 2-lists or 2-tuples), e.g.

```
class ALGraph:
    def __init__(self, verts=None, edges=None, is_directed=True):
        self.adjacency_list = {}
        self.addVerts(verts)
        self.addEdges(edges)
        self.is_directed = is_directed

    def addVert(self, vert):
        if vert is not None:
            self.adjacency_list[vert] = (self.adjacency_list[vert] is None) ? set() : self.adjacency_list[vert]

    def __addDirectedEdge(self, edge):
        self.adjacency_list[edge[0]].add(edge[1])

    def addEdge(self, edge):
        # could force list of verts to be only those passed in explicitly as vertices, in which case you'd want to check that both source and sink are legal verts
        # however, let's make the set of legal verts the union of things originally passed as verts AND those mentioned as endpoints of edges
        if edge is not None and len(edge) > 1:
            self.addVert(edge[0])
            self.addVert(edge[1])
            self.__addDirectedEdge(edge[0], edge[1])
            if not self.is_directed:
                 self.__addDirectedEdge(edge[1], edge[0])

    def addVerts(self, verts):
        if verts is not None:
            for vert in verts:
                self.addVert(vert)

    def addEdges(self, edges):
        if edges is not None:
            for edge in edges:
                self.addEdge(edge[0], edge[1])
```
I'm sure there's a typo or bug in there somewhere, but you get the idea -- anyway, given the above, rewrite findTopologicalOrdering() to work correctly?